### DISCUSSION TITLE
Does LineLength Have AutoCorrect?

### DISCUSSION THREAD

**Discussion by ericadohring** [detekt/detekt#4049] [Q&A] [answered]
Detekt is not auto correcting line length for me running it through gradle. My build.gradle is set up with  `autoCorrect = true` and autocorrect seems to work fine for other rules.

**Comment by cortinico** [comment, marked as answer]
Which rule are you referring to specifically? Detekt has two that you might refer to:
- `MaxLineLength` which does NOT have `autoCorrect` support.
- `MaximumLineLength` (a rule from `detekt-formatting`) which has `autoCorrect` support

**Comment by ericadohring** [comment]
I have MaximumLineLength in my detekt-config.yml - running ./gradlew detekt points out the errors, but doesn't fix them

```
  MaximumLineLength:
    active: true
    maxLineLength: 120
```

build.gradle
  ```
  detekt {
        autoCorrect = true
        config = files("pathto...../detekt-config.yml")
...
}

```

Just to make sure autocorrect was getting picked up, I ensured that autocorrected (it did)
```
  ImportOrdering:
    active: true
    layout: .....
```

**Reply by cortinico** [reply]
I've just checked the KtLint source code here: https://github.com/pinterest/ktlint/blob/master/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/MaxLineLengthRule.kt

It seems like, although the rule is accepting an `autoCorrect` parameter in input, that parameter is ignored. I believe the rationale is that the rule can't know where your long line should be split.

**Reply by cortinico** [reply]
You could consider asking the same question in the KtLint issue board 👍
